### PR TITLE
API Blueprint

### PR DESCRIPTION
**Pull request by ouziel-slama**

### CODE REVIEW

File: docs/advanced/api/rest.md [hunk 1]
```diff
@@ -2,161 +2,189 @@
...
+            "expiration": 2000,
+            "expire_index": 312010,
+            "fee_required": 900000,
+            "fee_required_remaining": 900000,
+            "fee_provided": 6800,
+            "fee_provided_remaining": 6800,
+            "status": "open"
+        }
+
+### View an Order Matches [/orders/{tx_hash}/matches]
```
Comment (adamkrellenstein) [inline docs/advanced/api/rest.md]: This path seems a bit unintuitive...

File: docs/advanced/api/rest.md [hunk 2]
```diff
@@ -2,161 +2,189 @@
...
++ expiration
++ expire_index
++ fee_required
++ fee_required_remaining
++ fee_provided
++ fee_provided_remaining
++ status
+
++ Parameters
+    + tx_hash: "1899b2e6ec36ba4bc9d035e6640b0a62b08c3a147c77c89183a77d9ed9081b3a" (required, string) - tx_hash of the order
```
Comment (adamkrellenstein) [inline docs/advanced/api/rest.md:136]: "`tx_hash` of the order" should maybe be "hash of the transaction that created the order" to be more understandable

File: docs/advanced/api/rest.md [hunk 3]
```diff
@@ -2,161 +2,189 @@
...
+            "expiration": 2000,
+            "expire_index": 312010,
+            "fee_required": 900000,
+            "fee_required_remaining": 900000,
+            "fee_provided": 6800,
+            "fee_provided_remaining": 6800,
+            "status": "open"
+        }
+
+### View an Order Match [/orders/{tx_hash}/matches]
```
Comment (adamkrellenstein) [inline docs/advanced/api/rest.md:162]: This path is not super obvious to me

File: docs/advanced/api/rest.md [hunk 4]
```diff
@@ -2,161 +2,189 @@
...
+                "status": "open"
+            },
+            ...
+        ]
+
+# Addresses
+
+Resources related to addresses in the API.
+
+### View an Address Balances [/assets/{address}/balances]
```
Comment (adamkrellenstein) [inline docs/advanced/api/rest.md:95]: "View Balances of a Given Asset"

File: docs/advanced/api/rest.md (hunk 1 above)
Comment (ouziel-slama) [reply docs/advanced/api/rest.md]: yep! I replaced everywhere `tx_hash` by `order_hash`, `bet_hash` and `dispenser_hash`. Now `tx_hash` in used only for `/transactions` routes.

File: docs/advanced/api/rest.md (hunk 2 above)
Comment (ouziel-slama) [reply docs/advanced/api/rest.md]: done!

File: docs/advanced/api/rest.md (hunk 3 above)
Comment (ouziel-slama) [reply docs/advanced/api/rest.md]: same!

File: docs/advanced/api/rest.md (hunk 4 above)
Comment (ouziel-slama) [reply docs/advanced/api/rest.md]: Now it's "Get Balances By Address". This title is automatically generated from the function name `get_balance_by_address()`.

File: docs/advanced/api-v2/rest.md [hunk 5]
```diff
@@ -0,0 +1,4019 @@
...
+- [`/transactions`](#group-transactions)
+- [`/addresses`](#group-addresses)
+- [`/assets`](#group-assets)
+- [`/orders`](#group-orders)
+- [`/bets`](#group-bets)
+- [`/dispensers`](#group-dispensers)
+- [`/burns`](#group-burns)
+- [`/events`](#group-events)
+- [`/mempool`](#group-mempool)
+- [`/backend`](#group-backend)
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:23]: I think we should change this to `/bitcoin/.

File: docs/advanced/api-v2/rest.md [hunk 6]
```diff
@@ -0,0 +1,4019 @@
...
+
+Notes:
+
+- When the server is not ready, that is to say when all the blocks are not yet parsed, all routes return a 503 error except `/` and those in the `/blocks`, `/transactions` and `/backend` groups which always return a result.
+
+- All API responses contain the following 3 headers:
+
+    * `X-COUNTERPARTY-HEIGHT` contains the last block parsed by Counterparty
+    * `X-BACKEND-HEIGHT` contains the last block known to Bitcoin Core
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:32]: `X-BITCOIN-HEIGHT`

... and everywhere else :)

File: docs/advanced/api-v2/rest.md [hunk 7]
```diff
@@ -0,0 +1,4019 @@
...
+- All API responses follow the following format:
+
+    ```
+    {
+        "error": <error_messsage_if_success_is_false>,
+        "result": <result_of_the_query_if_success_is_true>
+    }
+    ```
+
+- Routes in the `/backend` group serve as a proxy to make requests to AddrindexRS.
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:44]: to Bitcoin Core

File: docs/advanced/api-v2/rest.md [hunk 8]
```diff
@@ -0,0 +1,4019 @@
...
+
+Notes:
+
+- When the server is not ready, that is to say when all the blocks are not yet parsed, all routes return a 503 error except `/` and those in the `/blocks`, `/transactions` and `/backend` groups which always return a result.
+
+- All API responses contain the following 3 headers:
+
+    * `X-COUNTERPARTY-HEIGHT` contains the last block parsed by Counterparty
+    * `X-BACKEND-HEIGHT` contains the last block known to Bitcoin Core
+    * `X-COUNTERPARTY-READY` contains true if `X-COUNTERPARTY-HEIGHT` >= `X-BACKEND-HEIGHT` - 1
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:33]: What happens if bitcoind is still catching up?

File: docs/advanced/api-v2/rest.md [hunk 9]
```diff
@@ -0,0 +1,4019 @@
...
+                {
+                    "event_index": 14194760,
+                    "event": "BLOCK_PARSED",
+                    "bindings": {
+                        "block_index": 840464,
+                        "ledger_hash": "b3f8cbb50b0705a5c4a8495f8b5128de13a32daebd8ac5e8316a010f0d203584",
+                        "messages_hash": "801d961c45a257f85ef0f10a6a8fdf048a520ae4861c0903f26365b3eaaaf540",
+                        "txlist_hash": "84bdc5b9073f775a2b65de7da2b10b89a2235f3501883b0a836e41e68cd00d46"
+                    },
+                    "block_index": 840464,
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:183]: I'd skip `block_index` and `timestamp`. The index is in the query, and both are the same for every entry.

File: docs/advanced/api-v2/rest.md [hunk 10]
```diff
@@ -0,0 +1,4019 @@
...
+
++ Response 200 (application/json)
+
+    ```
+        {
+            "result": [
+                {
+                    "event_index": 14194760,
+                    "event": "BLOCK_PARSED",
+                    "bindings": {
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:177]: I think we should take the opportunity to rename this... maybe to `parameters` or `details`?

File: docs/advanced/api-v2/rest.md [hunk 11]
```diff
@@ -0,0 +1,4019 @@
...
+                        "previous_block_hash": "00000000000000000002db1e5aa19784eec3de949f98ec757e7a7f2fc392079d"
+                    },
+                    "block_index": 840464,
+                    "timestamp": 1713852779
+                }
+            ]
+        }
+    ```
+
+### Get Events Counts By Block [GET `/blocks/{block_index}/events/counts`]
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:300]: *Event Counts

File: docs/advanced/api-v2/rest.md [hunk 12]
```diff
@@ -0,0 +1,4019 @@
...
+        }
+    ```
+
+### Unpack [GET `/transactions/unpack`]
+
+Unpacks Counterparty data in hex format and returns the message type and data.
+
++ Parameters
+    + datahex: `16010b9142801429a60000000000000001000000554e4e45474f544941424c45205745204d555354204245434f4d4520554e4e45474f544941424c4520574520415245` (str, required) - Data in hex format 
+    + block_index (int, optional) - Block index of the transaction containing this data
```
Comment (adamkrellenstein) [inline docs/advanced/api-v2/rest.md:695]: why is this here? shouldn't be any faster...

**Review by adamkrellenstein — APPROVED** [review]
Nice!!! minor comments only :)

File: docs/advanced/api-v2/rest.md [hunk 13]
```diff
@@ -0,0 +1,4019 @@
...
+                    "tx_hash": "2c8bc3eede9ec60d26c6fd7f44829adc64da593552044a28c673022220f560c3"
+                },
+                {
+                    "tx_hash": "a209e345549cffef6e2190b53ac0222afc965fd618843df5ccbd645a6a7999ee"
+                }
+            ]
+        }
+    ```
+
+### Get Oldest Tx [GET `/backend/addresses/{address}/transactions/oldest`]
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:3714]: get_oldest_transaction_by_address() ?

File: docs/advanced/api-v2/rest.md (hunk 5 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:23]: done!

File: docs/advanced/api-v2/rest.md (hunk 6 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:32]: done!

File: docs/advanced/api-v2/rest.md (hunk 7 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:44]: done

File: docs/advanced/api-v2/rest.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (adamkrellenstein) [reply docs/advanced/api-v2/rest.md:3714]: sure!

File: docs/advanced/api-v2/rest.md (hunk 8 above)
Comment (adamkrellenstein) [reply docs/advanced/api-v2/rest.md:33]: ach sorry I don't follow. can you say that again?

Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:33]: No problem what I said is wrong anyway ;-) for now as long as Addrindexrs is catching up the API server is not starting..

File: docs/advanced/api-v2/rest.md [hunk 14]
```diff
@@ -0,0 +1,4019 @@
...
+        {
+            "result": {
+                "status": "Healthy"
+            }
+        }
+    ```
+
+## Group Backend
+
+### Search Raw Transactions [GET `/backend/addresses/{address}/transactions`]
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:3658]: done

File: docs/advanced/api-v2/rest.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:3714]: done

File: docs/advanced/api-v2/rest.md [hunk 15]
```diff
@@ -0,0 +1,4019 @@
...
+
++ Response 200 (application/json)
+
+    ```
+        {
+            "result": "0388ef0905568d425f1ffd4031d93dda4ef0e220c9b5fc4a6cbaf11544c4a5ca49"
+        }
+    ```
+
+### Get Raw Transaction [GET `/backend/transactions/{tx_hash}`]
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:3851]: done

**Comment by droplister** [comment]
Is there a way to introduce name and quantity normalization on all API results in a way that's still performant and maintainable?

It seems like there could be a way to join `asset_longname` and `divisible` fields with most queries to support normalization but is there a way to perform the normalization on top of that?

**Name**
For example, if the `asset` is always/only returned it's necessary to check if an `asset_longname` exists for display purposes due to how subassets have been implemented.

In practice, I find myself doing a lot of querying/caching/mapping to achieve this. Maybe there's a DRY method to join  `asset_longname` from the **Assets** table with all queries in the [ledger.py](https://github.com/CounterpartyXCP/counterparty-core/blob/0b77eb9f38281d69ce2e08e740bc8d0db3f29462/counterparty-core/counterpartycore/lib/ledger.py)?

I think the ideal would be to offer something like an `asset_normalized`, so that developers don't need to write their own conditional ternary operator as a utility function that's used everywhere.

```
    {
        "result": [
            {
                 ...
                "asset": "A1976569558204240008",
                "asset_longname": "BITCORN.GLASSES" // joined
                ...
            },
            {
                 ...
                "asset": "A1976569558204240008",
                "asset_normalized": "BITCORN.GLASSES" // computed from join
                ...
            },
        ]
    }
```

*If this did not get added, I would probably index the /assets endpoint and map this field onto API results within a caching layer.*

**Quantity**
Normalizing quantities is another big quality of life improvement that's worth thinking about. The same **Assets** table join that is used for selecting `asset_longname` could also select `divisible`.

That would allow for a utility function to format quantities for display by passing the quantity and a divisibility boolean.

However, like with asset names, I think the ideal is that this would be done by Counterparty. It's more computation than a join and may not make sense, but it would help a great deal.

```
    {
        "result": [
            {
                 ...
                "quantity": 100000000,
                "asset_divisible": true // joined
                ...
            },
            {
                 ...
                "quantity": 100000000,
                "quantity_normalized": 1.00000000 // computed from join
                ...
            },
        ]
    }
```

*Joining asset_longname and divisible would be plenty helpful, but normalization would be epic. (Needs some thought around naming for things like order matches with two assets.)*

**Comment by droplister** [comment]
Related to quantity normalization, because you need to normalize quantities before calculating an exchange rate, are prices. 

**Exchange Rates: Orders/Order Matches**
This is something that Counterblock used to do, but might not make sense to spend time on right now. If we maintain an array of [quote assets](https://github.com/CounterpartyXCP/counterblock/blob/0bf2a190811672d0d4b3785e0da35168635c3461/counterblock/lib/config.py#L35C1-L35C14), we could compute exchange rates on order and order match API results.

```
    {
        "result": [
            {
                "give_asset": "PEPECASH",
                "give_quantity": 6966600000000,
                "give_quantity_normalized": 69666.00000000,
                "give_remaining": 900000000000,
                "give_remaining_normalized": 9000.00000000,
                "get_asset": "XCP",
                "get_quantity": 11076894000,
                "get_quantity_normalized": 110.76894000,
                "get_remaining": 1431000000,
                "get_remaining_normalized": 14.31000000,
                "market_pair": "PEPECASH/XCP",
                "market_dir": "Sell",
                "market_rate": "628.9308176100629", // What level of precision here can be tricky to decide 
            }
        ]
    }
```
*It's not necessary to maintain a quote asset array to improve on this for developers, but it does help and once you do its easy to decide whether its a sell or a buy order.*

**Exchange Rates: Dispensers/Dispenses**
It's also something that could be improved for dispenses and dispensers too. In this case, the quote asset is always BTC. But you need to lookup the dispenser on a dispense to see what the satoshirate and dispense quantity were. And if the quantity dispensed is greater than 1 you need to do an additional calculation to determine what the effective price per unit is.

Also, from the dispense alone it's not clear whether the dispense triggered more than one dispense from the dispenser. So, a lot of lookups and checks start to be necessary to make these trades understandable.

```
    {
        "result": [
            {
                "tx_index": 2726580,
                "dispense_index": 0,
                "tx_hash": "e7f0f2c9bef7a492b714a5952ec61b283be344419c5bc33f405f9af41ebfa48b",
                "block_index": 840322,
                "source": "bc1qq735dv8peps2ayr3qwwwdwylq4ddwcgrpyg9r2",
                "destination": "bc1qzcdkhnexpjc8wvkyrpyrsn0f5xzcpu877mjmgj",
                "asset": "FLOCK",
                "dispense_quantity": 90000000000,
                "dispense_quantity_normalized": 900.00000000,
                "dispenser_tx_hash": "753787004d6e93e71f6e0aa1e0932cc74457d12276d53856424b2e4088cc542a"
                "number_of_dispenses": 12,
                "sats_per_dispense": 12356,
                "sats_per_unit": 12345 
            }
        ]
    }
```

I think it's reasonable to expect explorers and apps to compute some of these things for themselves, but if Counterparty API takes on some of this or at least joins the fields that are needed to perform these normalizations and calculations you'll get fewer apps re-implementing and maintaining their own state of the network.

In a perfect world, marketplace explorers are figuring out volume and trading prices in dollar terms and aggregating that type of information in their databases.

File: docs/advanced/api-v2/rest.md (hunk 11 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:300]: fixed

File: docs/advanced/api-v2/rest.md [hunk 16]
```diff
@@ -0,0 +1,4019 @@
...
+    + wager_quantity: `1000` (int, required) - The quantities of XCP to wager (in satoshis, hence integer) 
+    + counterwager_quantity: `1000` (int, required) - The minimum quantities of XCP to be wagered against, for the bets to match 
+    + expiration: `100` (int, required) - The number of blocks after which the bet expires if it remains unmatched 
+    + leverage (int, optional) - Leverage, as a fraction of 5040
+        + Default: `5040`
+    + target_value: `1000` (int, optional) - Target value for Equal/NotEqual bet 
+        + Default: `None`
+    + encoding (str, optional) - The encoding method to use
+        + Default: `auto`
+    + fee_per_kb (int, optional) - The fee per kilobyte of transaction data constant that the server uses when deciding on the dynamic fee to use (in satoshi)
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:1243]: fixed

File: docs/advanced/api-v2/rest.md [hunk 17]
```diff
@@ -0,0 +1,4019 @@
...
+    + dust_return_pubkey (str, optional) - The dust return pubkey is used in multi-sig data outputs (as the only real pubkey) to make those the outputs spendable. By default, this pubkey is taken from the pubkey used in the first transaction input. However, it can be overridden here (and is required to be specified if a P2SH input is used and multisig is used as the data output encoding.) If specified, specify the public key (in hex format) where dust will be returned to so that it can be reclaimed. Only valid/useful when used with transactions that utilize multisig data encoding. Note that if this value is set to false, this instructs counterparty-server to use the default dust return pubkey configured at the node level. If this default is not set at the node level, the call will generate an exception
+        + Default: `None`
+    + disable_utxo_locks (bool, optional) - By default, UTXO's utilized when creating a transaction are 'locked' for a few seconds, to prevent a case where rapidly generating create_ calls reuse UTXOs due to their spent status not being updated in bitcoind yet. Specify true for this parameter to disable this behavior, and not temporarily lock UTXOs
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:1263]: fixed

File: docs/advanced/api-v2/rest.md [hunk 18]
```diff
@@ -0,0 +1,4019 @@
...
+        + Default: `False`
+    + extended_tx_info (bool, optional) - When this is not specified or false, the create_ calls return only a hex-encoded string. If this is true, the create_ calls return a data object with the following keys: tx_hex, btc_in, btc_out, btc_change, and btc_fee
+        + Default: `False`
+    + p2sh_pretx_txid (str, optional) - The previous transaction txid for a two part P2SH message. This txid must be taken from the signed transaction
+        + Default: `None`
+    + old_style_api (bool, optional) - Use the old style API
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:1269]: yeah I removed this one! (I just copy/past form the old doc)

File: docs/advanced/api-v2/rest.md [hunk 19]
```diff
@@ -0,0 +1,4019 @@
...
+                    "tx1_expiration": 8064,
+                    "match_expire_index": 840401,
+                    "fee_paid": 0,
+                    "status": "completed"
+                }
+            ]
+        }
+    ```
+
+### Get Btcpays By Order [GET `/orders/{order_hash}/btcpays`]
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:2949]: fixed

File: docs/advanced/api-v2/rest.md [hunk 20]
```diff
@@ -0,0 +1,4019 @@
...
+                    "value": 12354,
+                    "confirmations": 1621,
+                    "amount": 0.00012354,
+                    "txid": "23758832e0fc92a7ea303623b8f743219cb8e637e7e7ac9fb6f90641efac9379"
+                }
+            ]
+        }
+    ```
+
+### Pubkeyhash To Pubkey [GET `/backend/addresses/{address}/pubkey`]
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:3834]: fixed

File: docs/advanced/api-v2/rest.md [hunk 21]
```diff
@@ -0,0 +1,4019 @@
...
+                    "block_index": 744232,
+                    "timestamp": 1712256340
+                }
+            ]
+        }
+    ```
+
+## Group Healthz
+
+### Check Server Status [GET `/healthz`]
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:3638]: LOL! fixed!

File: docs/advanced/api-v2/rest.md (hunk 12 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:695]: This is to manage protocol changes.. if you want to unpack a transaction before a given change..

File: docs/advanced/api-v2/rest.md [hunk 22]
```diff
@@ -0,0 +1,4019 @@
...
+                    "locked": 0,
+                    "status": "valid",
+                    "asset_longname": null,
+                    "reset": 0
+                }
+            ]
+        }
+    ```
+
+### Get Sends Or Receives By Block [GET `/blocks/{block_index}/sends`]
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:549]: yeah awkward! fixed!

File: docs/advanced/api-v2/rest.md (hunk 10 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:177]: payload ?

File: docs/advanced/api-v2/rest.md [hunk 23]
```diff
@@ -0,0 +1,4019 @@
...
+        + Default: `None`
+    + encoding (str, optional) - The encoding method to use
+        + Default: `auto`
+    + fee_per_kb (int, optional) - The fee per kilobyte of transaction data constant that the server uses when deciding on the dynamic fee to use (in satoshi)
+        + Default: `None`
+    + regular_dust_size (int, optional) - Specify (in satoshi) to override the (dust) amount of BTC used for each non-(bare) multisig output.
+        + Default: `546`
+    + multisig_dust_size (int, optional) - Specify (in satoshi) to override the (dust) amount of BTC used for each (bare) multisig output
+        + Default: `1000`
+    + op_return_value (int, optional) - The value (in satoshis) to use with any OP_RETURN outputs in the generated transaction. Defaults to 0. Don't use this, unless you like throwing your money away
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:1249]: :-) yeah! most of this optionals parameters are probably never used!
will remove this one to start!

File: docs/advanced/api-v2/rest.md (hunk 9 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:183]: done

File: docs/advanced/api-v2/rest.md (hunk 23 above)
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:1249]: done

File: docs/advanced/api-v2/rest.md (hunk 10 above)
Comment (adamkrellenstein) [reply docs/advanced/api-v2/rest.md:177]: I prefer `params` I think.

File: docs/advanced/api-v2/rest.md (hunk 8 above)
Comment (adamkrellenstein) [reply docs/advanced/api-v2/rest.md:33]: Ahh I think we want it to run either way, no? It'd be nice if it gave a helpful error while bitcoind is stupidly completely unavailable.

Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:33]: yeah I agree.. but it's not trivial.. the problem comes from Addrindexrs which does not start the RPC server until it has caught up.. this means that we cannot I don't even have the block count.. and also the connection step to Addrindexrs is blocking.. I will try to find a clean solution..

File: docs/advanced/api-v2/rest.md (hunk 10 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:177]: done!

File: docs/advanced/api-v2/rest.md [hunk 24]
```diff
@@ -0,0 +1,4019 @@
...
+    + bet_type: `2` (int, required) - Bet 0 for Bullish CFD (deprecated), 1 for Bearish CFD (deprecated), 2 for Equal, 3 for NotEqual 
+    + deadline: `3000000000` (int, required) - The time at which the bet should be decided/settled, in Unix time (seconds since epoch) 
+    + wager_quantity: `1000` (int, required) - The quantities of XCP to wager (in satoshis, hence integer) 
+    + counterwager_quantity: `1000` (int, required) - The minimum quantities of XCP to be wagered against, for the bets to match 
+    + expiration: `100` (int, required) - The number of blocks after which the bet expires if it remains unmatched 
+    + leverage (int, optional) - Leverage, as a fraction of 5040
+        + Default: `5040`
+    + target_value: `1000` (int, optional) - Target value for Equal/NotEqual bet 
+        + Default: `None`
+    + encoding (str, optional) - The encoding method to use
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md:1241]: Added here https://github.com/CounterpartyXCP/counterparty-core/pull/1729/commits/9f686bf9f6e59b75466e50b2cd9847ef46334e27

**Comment by ouziel-slama** [comment]
@droplister yes it can actually be interesting! I need to do some performance testing to find the best way to do this.
I opened an issue for this: https://github.com/CounterpartyXCP/counterparty-core/issues/1731

File: docs/advanced/api-v2/rest.md (hunk 8 above)
Comment (ouziel-slama) [reply docs/advanced/api-v2/rest.md]: ok done here https://github.com/CounterpartyXCP/counterparty-core/pull/1729/commits/9d61360bad2af407fa9e504948bef9c18a2f31bc
Now the API v2 starts before connecting to addrindexrs.. and as long it's not connected it returns a 503 error with a "Backend still not ready" message.
